### DISCUSSION TITLE
Channel access: View entries, by others

### DISCUSSION THREAD

**Discussion by shbchk** [ExpressionEngine/ExpressionEngine#3091] [Feature Requests]
Not sure if this is a bug, but there is some inconsistency for sure.

1. Members with the role "OwnEntriesEditors" can edit only entries they created. In the entry list, they can see their own entries only.
2. But pagination suggests that there are a lot of other entries https://github.com/ExpressionEngine/ExpressionEngine/issues/2178
3. And they can see entries created by others in the Relationship fields (if it's enabled).

I wonder if we add an additional action **"View entries, by others"** here?
![2c56916e-2591-4b5b-ad80-4a8e3fd3bf0a](https://user-images.githubusercontent.com/12234507/223495511-bab4bf09-1b3b-4355-a257-d4525644965a.png)

**Comment by intoeetive** [comment]
@TomJaeger what do you think about this idea (or we could just list entries without edit link, even if the people don't have edit access)

**Comment by shbchk** [comment]
Listing all the entries might be not a good thing - one can have channels with restricted access

**Comment by intoeetive** [comment]
This is actually how it works already... If you have "edit entries by others" checked for one channel, you can see (but not edit) entries in all other channels (where you can edit your own entries)

If you don't have "edit entries by others" checked anywhere, then you can't see the entries.

I think we need to make this consistent at the minimum. 

Considering it a bug and moving to issues.

Thanks for reporting!

**Comment by intoeetive** [comment]
https://github.com/ExpressionEngine/ExpressionEngine/issues/3099

**Comment by TomJaeger** [comment]
Just to confirm @intoeetive when you say consistent... you mean consistent between if someone has "edit entries by others" for one channel, and being able to see them, should be consistent with if that's not checked at all?

If so, makes sense to me... which way do you think is best for consistency?

**Comment by intoeetive** [comment]
@TomJaeger sorry for not being 100% clear... I think we need to let people see **all** entries in channel if they have at least any permission in that channel. Which is what is happening now if you have "edit own entries" permission in one channel and "edit entries by others" in any other channel. I think they should be able to see the entries in channel they have access to, even if they don't have "edit entries by others" permission anywhere at all.

**Comment by intoeetive** [comment]
@TomJaeger assigned you to review this PR https://github.com/ExpressionEngine/ExpressionEngine/pull/3196

**Comment by intoeetive** [comment]
We had this reported in Slack:

> I noticed an issue after updating to version 7.3.4 where certain members were able to view entries they shouldn’t have been able to. Despite the role settings only allowing for editing their own, this specific role was able to view all entries. While this is not harmful, but the entry listing is not what we used to have

**Comment by intoeetive** [comment]
one of possible solutions that came to my mind is that we can have "This channel is private" setting (toggle) on channel.
Default is off, entries are public and everyone can see them - on frontend and also the editors that have some level of access to channel in CP.
If this setting is on (channel is private) then we show selection of the roles that can VIEW entries in this channel. If their role is not in the list, user cannot see entry - neither in CP not in `exp:channel:entries`. This will help people to easier build sites with restricted access to data on front-end (now you can only manage data access in the templates). In the relationship field, if the entry is selected and the user does not have access to it, we'll replace the title with something like `[private entry]`

**Reply by intoeetive** [reply]
Another option would be to just pre-select filter by author when clicking channel in navigation if they only have access to own entries. That way, they would still see all entries, but the default list would be smaller.

**Comment by robinsowell** [comment]
1. I do like an additional setting- probably 'View entry titles by others', because people use EE in ways I'd never consider and that might be useful.
2. I personally can't think of any reason I'd want to see entries I can't edit crowding up the entry editor.  But the fact I can't imagine it doesn't mean it doesn't exist.

I can think of lots of reasons that a person assigning related entries might want to see entries that they didn't write.  Here's where I find the idea of marking a channel 'private' interesting.  Meh.  I need to think more.

**Comment by intoeetive** [comment]
Implemented via https://github.com/ExpressionEngine/ExpressionEngine/pull/3196
